src/components/PurchaseOrder/PO.js

```diff
@@ -188,6 +189,10 @@ const PO = ({
     !checkinItems
     && (cost?.quantityPhysical || 0 + cost?.quantityElectronic || 0) > 0
   ));
+  const unopenMessage = hasRemovablePieces ? 'withPieces' : 'withoutPieces';
+  const isReceiptNotRequired = poLines?.every(({ receiptStatus }) => (
```

Comment:
hm, looks like you need
```
const isReceiptNotRequired = poLines?.every(({ receiptStatus }) => (
...
const hasRemovablePieces = prev || isReceiptNotRequired
```
instead (or smth like that) instead of creating `unopenMessage` and selecting  withPieces/withoutPieces several times

Reply:
This approach looks much better. Thanks. Updated conditions